app/src/main/java/com/spendster/presentation/addExpenses/chooseCategory/CategoryRecyclerViewAdapter.java

```diff
@@ -0,0 +1,82 @@
+package com.spendster.presentation.addExpenses.chooseCategory;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.CardView;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.spendster.R;
+import com.spendster.data.entity.Category;
+
+import java.util.List;
+
+public class CategoryRecyclerViewAdapter extends RecyclerView.Adapter<CategoryRecyclerViewAdapter.CategoryViewHolder> {
+
+    private final Context context;
+    private List<Category> categories;
+    private final OnCategoryClickListener onCategoryClickListener;
+
+    public CategoryRecyclerViewAdapter(Context context, List<Category> categories, OnCategoryClickListener onCategoryClickListener) {
+        this.context = context;
+        this.categories = categories;
+        this.onCategoryClickListener = onCategoryClickListener;
+    }
+
+    @NonNull
+    @Override
+    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        View view = LayoutInflater.from(context).inflate(R.layout.category, viewGroup, false);
+        final CategoryViewHolder categoryViewHolder = new CategoryViewHolder(view);
+        categoryViewHolder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                onCategoryClickListener.onCategoryClick(categories.get(categoryViewHolder.getLayoutPosition()));
+            }
+        });
+        return categoryViewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull CategoryViewHolder categoryViewHolder, int i) {
+        categoryViewHolder.bind(categories.get(i));
+    }
+
+    @Override
+    public int getItemCount() {
+        int size;
+        if (categories == null) {
+            size = 0;
+        } else {
+            size = categories.size();
+        }
+        return size;
+    }
+
+    public void bind(List<Category> categories){
+        this.categories = categories;
+        notifyItemRangeInserted(0, categories.size());
```

Comment:
@SimonMaha what if categories here is null? 
please use getItemCount() method as second param here